actions/notification_actions.jsx

```diff
@@ -17,121 +17,132 @@ import {stripMarkdown} from 'utils/markdown';
+                name: msgProps.channel_name,
+                type: msgProps.channel_type,
+            };
+        } else if (channel.type === Constants.DM_CHANNEL) {
+            title = Utils.localizeMessage('notification.dm', 'Direct Message');
+        } else {
+            title = channel.display_name;
+        }
+
+        if (title === '') {
+            if (msgProps.channel_type === Constants.DM_CHANNEL) {
+                title = Utils.localizeMessage('notification.dm', 'Direct Message');
+            } else {
+                title = msgProps.channel_display_name;
+            }
         }
-    }
-
-    let notifyText = post.message;
-
-    const msgPropsPost = JSON.parse(msgProps.post);
-    const attachments = msgPropsPost && msgPropsPost.props && msgPropsPost.props.attachments ? msgPropsPost.props.attachments : [];
-    let image = false;
-    attachments.forEach((attachment) => {
-        if (notifyText.length === 0) {
-            notifyText = attachment.fallback ||
-                         attachment.pretext ||
-                         attachment.text;
+
+        let notifyText = post.message;
+
+        const msgPropsPost = JSON.parse(msgProps.post);
+        const attachments = msgPropsPost && msgPropsPost.props && msgPropsPost.props.attachments ? msgPropsPost.props.attachments : [];
+        let image = false;
+        attachments.forEach((attachment) => {
+            if (notifyText.length === 0) {
+                notifyText = attachment.fallback ||
+                    attachment.pretext ||
+                    attachment.text;
+            }
+            image |= attachment.image_url.length > 0;
+        });
+
+        let strippedMarkdownNotifyText = stripMarkdown(notifyText);
+        if (strippedMarkdownNotifyText.length > NOTIFY_TEXT_MAX_LENGTH) {
+            strippedMarkdownNotifyText = strippedMarkdownNotifyText.substring(0, NOTIFY_TEXT_MAX_LENGTH - 1) + '...';
         }
-        image |= attachment.image_url.length > 0;
-    });
-
-    let strippedMarkdownNotifyText = stripMarkdown(notifyText);
-    if (strippedMarkdownNotifyText.length > NOTIFY_TEXT_MAX_LENGTH) {
-        strippedMarkdownNotifyText = strippedMarkdownNotifyText.substring(0, NOTIFY_TEXT_MAX_LENGTH - 1) + '...';
-    }
-
-    let body = '';
-    if (strippedMarkdownNotifyText.length === 0) {
-        if (msgProps.image) {
-            body = username + Utils.localizeMessage('channel_loader.uploadedImage', ' uploaded an image');
-        } else if (msgProps.otherFile) {
-            body = username + Utils.localizeMessage('channel_loader.uploadedFile', ' uploaded a file');
-        } else if (image) {
-            body = username + Utils.localizeMessage('channel_loader.postedImage', ' posted an image');
+
+        let body = `@${username}: `;
+        if (strippedMarkdownNotifyText.length === 0) {
+            if (msgProps.image) {
+                body += Utils.localizeMessage('channel_loader.uploadedImage', ' uploaded an image');
```

Comment:
These are going to end up as `@user:  uploaded an image`. We should get rid of the colon and extra space

Reply:
this follows the push notification spec

Reply:
Is that the spec? I just checked on my phone, and it has "@User attached a file."

Reply:
Thanks. I'm all about obsessively correct use of punctuation 😛